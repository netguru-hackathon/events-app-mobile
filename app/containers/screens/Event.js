import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Dimensions,
  Image,
  StyleSheet,
  Text,
  ScrollView,
  View,
} from 'react-native';
import I18n from '../../utils/translations';
import { fetchEvent } from '../../actions/events';
import { RenderActivityIndicator } from '../shared/RenderActivityIndicator';
import colors from '../../constants/colors';
import EventItems from './EventItems';

const { width, height } = Dimensions.get('window');
const styles = StyleSheet.create({
  descriptionContainer: {
    padding: 10,
  },
  icon: {
    width: 20,
    height: 20,
  },
  image: {
    marginTop: 0,
    width,
    height: 0.25 * height,
  },
});

class Event extends Component {
  static propTypes = {
    fetchEvent: PropTypes.func.isRequired,
    isStarted: PropTypes.bool.isRequired,
    isFetching: PropTypes.bool.isRequired,
    event: PropTypes.shape({
      name: PropTypes.string,
      description: PropTypes.string,
      image: PropTypes.string,
    }).isRequired,
    navigation: PropTypes.shape({
      state: PropTypes.object.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  static navigationOptions = (props) => {
    const { navigation } = props;

    return {
      tabBarLabel: I18n.t('events'),
      showIcon: true,
      tabBarIcon: ({ tintColor }) => (
        <Image
          source={require('../../img/EventsIcon.png')}
          style={[styles.icon, { tintColor }]}
        />
      ),
      headerTitle: `${navigation.state.params.item.name}`,
      headerTintColor: colors.WHITE,
      headerStyle: {
        position: 'absolute',
        backgroundColor: colors.BLACK_OPACITY,
        zIndex: 100,
        top: 0,
        left: 0,
        right: 0,
      },
    };
  };

  componentWillMount() {
    const { navigation } = this.props;

    this.props.fetchEvent(navigation.state.params.item.id);
  }

  // eslint-disable-next-line
  renderEvent() {
    const { isStarted, isFetching, event } = this.props;

    if (isStarted && isFetching) return <RenderActivityIndicator />;
    return (
      <View>
        <Image source={{ uri: event.image }} style={styles.image} />
        <View style={styles.descriptionContainer}>
          <Text>{event.description}</Text>
        </View>
      </View>
    );
  }

  renderEventItems() {
    const { isStarted, isFetching, event } = this.props;

    if (isStarted && isFetching) return <RenderActivityIndicator />;
    return <EventItems items={event.items} />;
  }

  render() {
    return (
      <ScrollView style={styles.container}>
        {this.renderEvent()}
        {this.renderEventItems()}
      </ScrollView>
    );
  }
}

const mapStateToProps = state => ({
  isStarted: state.event.isStarted,
  isFetching: state.event.isFetching,
  event: state.event.item,
});

const mapDispatchToProps = {
  fetchEvent,
};

export default connect(mapStateToProps, mapDispatchToProps)(Event);
