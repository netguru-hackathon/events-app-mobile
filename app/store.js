import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';

import axiosClient from './api/client';
import Reducers from './reducers/index';

const store = applyMiddleware(
  thunk,
  axiosMiddleware(axiosClient, { returnRejectedPromiseOnError: true }),
)(createStore)(Reducers);

export default store;
