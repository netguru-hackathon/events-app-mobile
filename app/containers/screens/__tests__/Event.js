import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Event from '../Event';
import store from '../../../store';

test('renders correctly', () => {
  const tree = renderer.create(
    <Event
      store={store}
      isStarted={'true'}
      isFetching={'true'}
      event={{
        name: 'name',
        description: 'description',
        image: 'image',
      }}
      navigation={{
        navigate: () => {},
        state: {
          params: {
            item: {
              id: 1,
              name: 'name',
            },
          },
        },
      }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
