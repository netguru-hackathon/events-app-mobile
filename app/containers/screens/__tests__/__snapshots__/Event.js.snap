// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={undefined}
>
  <View>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
          "padding": 8,
        }
      }
    />
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
          "padding": 8,
        }
      }
    />
  </View>
</RCTScrollView>
`;
